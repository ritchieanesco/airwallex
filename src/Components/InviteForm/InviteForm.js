import React, { Component, Fragment } from "react";
import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import { object, bool, func } from "prop-types";
import Text from "../Text/Text";
import Loader from "../Loader/Loader";

const styles = theme => ({
  button: {
    margin: theme.spacing.unit
  },
  buttonWrapper: {
    position: "relative"
  }
});

class InviteForm extends Component {
  handleSubmit = event => {
    event.preventDefault();
    const { name, email, confirmEmail } = event.currentTarget.elements;
    this.props.submitForm({
      name: name.value,
      email: email.value,
      confirmEmail: confirmEmail.value
    });
  };

  handleChange = event => {
    this.props.inputChange(event.currentTarget.name, event.currentTarget.value);
  };

  handleClose = event => {
    event.preventDefault();
    this.props.closeModal();
  };

  renderForm() {
    const { classes, errors, fields, isSubmitting, fail } = this.props;
    return (
      <form onSubmit={this.handleSubmit}>
        <TextField
          id="name"
          label="Name"
          name="name"
          InputLabelProps={{
            shrink: true
          }}
          placeholder="Name"
          fullWidth
          margin="normal"
          error={errors && errors.name ? errors.name : false}
          value={fields && fields.name ? fields.name : ""}
          onChange={e => this.handleChange(e)}
          helperText={
            errors && errors.name && "Name must be at least 3 characters long"
          }
        />
        <TextField
          id="email"
          label="Email"
          name="email"
          InputLabelProps={{
            shrink: true
          }}
          placeholder="Email"
          fullWidth
          margin="normal"
          error={errors && errors.email ? errors.email : false}
          value={fields && fields.email ? fields.email : ""}
          onChange={e => this.handleChange(e)}
          helperText={errors && errors.email && "Email must be a valid format"}
        />
        <TextField
          id="confirm-email"
          label="Confirmation Email"
          name="confirmEmail"
          InputLabelProps={{
            shrink: true
          }}
          placeholder="Confirmation Email"
          fullWidth
          margin="normal"
          error={errors && errors.confirmEmail ? errors.confirmEmail : false}
          value={fields && fields.confirmEmail ? fields.confirmEmail : ""}
          onChange={e => this.handleChange(e)}
          helperText={
            errors &&
            errors.confirmEmail &&
            "Confirm email does not match email field"
          }
        />

        <div className={classes.buttonWrapper}>
          {isSubmitting && <Loader size={28} />}
          <Button
            type="submit"
            className={classes.button}
            variant="raised"
            color="primary"
            disabled={isSubmitting}
          >
            Submit
          </Button>
        </div>
        {!isSubmitting &&
          fail && (
            <Text variant="body2">
              Sorry, there was an error sending your invitation. Please try
              again.
            </Text>
          )}
      </form>
    );
  }

  renderSuccess() {
    return (
      <div>
        <Text variant="subheading">Success</Text>
        <Text variant="body1">We will be in contact with you shortly</Text>
        <Button
          onClick={e => this.handleClose(e)}
          type="submit"
          color="primary"
        >
          Close
        </Button>
      </div>
    );
  }

  render() {
    const { success } = this.props;
    return (
      <Fragment>
        <Text variant="title" id="invite-modal-title">
          Request an Invite
        </Text>
        {success ? this.renderSuccess() : this.renderForm()}
      </Fragment>
    );
  }
}

InviteForm.propTypes = {
  classes: object,
  errors: object,
  fields: object,
  isSubmitting: bool,
  fail: bool,
  submitForm: func,
  inputChange: func,
  closeModal: func
};

export default withStyles(styles)(InviteForm);
