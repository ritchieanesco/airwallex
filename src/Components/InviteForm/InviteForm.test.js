import React from "react";
import renderer from "react-test-renderer";
import InviteForm from "./InviteForm";

describe("InviteForm", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<InviteForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
