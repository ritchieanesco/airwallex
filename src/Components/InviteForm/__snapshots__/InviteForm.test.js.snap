// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteForm should render correctly 1`] = `
Array [
  <h2
    className="MuiTypography-root-3 MuiTypography-title-9 MuiTypography-gutterBottom-20"
>
    Request an Invite
</h2>,
  <form
    onSubmit={[Function]}
>
    <div
        aria-describedby={undefined}
        className="MuiFormControl-root-27 MuiFormControl-marginNormal-28 MuiFormControl-fullWidth-30"
        onBlur={[Function]}
        onFocus={[Function]}
    >
        <label
            className="MuiFormLabel-root-36 MuiInputLabel-root-31 MuiInputLabel-formControl-32 MuiInputLabel-animated-35 MuiInputLabel-shrink-34"
            data-shrink={true}
            htmlFor="name"
        >
            Name
        </label>
        <div
            className="MuiInput-root-41 MuiInput-fullWidth-48 MuiInput-formControl-42 MuiInput-underline-45"
        >
            <input
                aria-invalid={false}
                aria-required={false}
                autoComplete={undefined}
                autoFocus={undefined}
                className="MuiInput-input-49"
                defaultValue={undefined}
                disabled={false}
                id="name"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder="Name"
                readOnly={undefined}
                required={undefined}
                rows={undefined}
                type="text"
                value=""
            />
        </div>
    </div>
    <div
        aria-describedby={undefined}
        className="MuiFormControl-root-27 MuiFormControl-marginNormal-28 MuiFormControl-fullWidth-30"
        onBlur={[Function]}
        onFocus={[Function]}
    >
        <label
            className="MuiFormLabel-root-36 MuiInputLabel-root-31 MuiInputLabel-formControl-32 MuiInputLabel-animated-35 MuiInputLabel-shrink-34"
            data-shrink={true}
            htmlFor="email"
        >
            Email
        </label>
        <div
            className="MuiInput-root-41 MuiInput-fullWidth-48 MuiInput-formControl-42 MuiInput-underline-45"
        >
            <input
                aria-invalid={false}
                aria-required={false}
                autoComplete={undefined}
                autoFocus={undefined}
                className="MuiInput-input-49"
                defaultValue={undefined}
                disabled={false}
                id="email"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder="Email"
                readOnly={undefined}
                required={undefined}
                rows={undefined}
                type="text"
                value=""
            />
        </div>
    </div>
    <div
        aria-describedby={undefined}
        className="MuiFormControl-root-27 MuiFormControl-marginNormal-28 MuiFormControl-fullWidth-30"
        onBlur={[Function]}
        onFocus={[Function]}
    >
        <label
            className="MuiFormLabel-root-36 MuiInputLabel-root-31 MuiInputLabel-formControl-32 MuiInputLabel-animated-35 MuiInputLabel-shrink-34"
            data-shrink={true}
            htmlFor="confirm-email"
        >
            Confirmation Email
        </label>
        <div
            className="MuiInput-root-41 MuiInput-fullWidth-48 MuiInput-formControl-42 MuiInput-underline-45"
        >
            <input
                aria-invalid={false}
                aria-required={false}
                autoComplete={undefined}
                autoFocus={undefined}
                className="MuiInput-input-49"
                defaultValue={undefined}
                disabled={false}
                id="confirm-email"
                name="confirmEmail"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder="Confirmation Email"
                readOnly={undefined}
                required={undefined}
                rows={undefined}
                type="text"
                value=""
            />
        </div>
    </div>
    <div
        className="InviteForm-buttonWrapper-2"
    >
        <button
            className="MuiButtonBase-root-70 MuiButton-root-54 MuiButton-raised-60 MuiButton-raisedPrimary-61 InviteForm-button-1"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="submit"
        >
            <span
                className="MuiButton-label-55"
            >
                Submit
            </span>
            <span
                className="MuiTouchRipple-root-73"
            />
        </button>
    </div>
</form>,
]
`;
